src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py

```diff
@@ -562,24 +562,68 @@ def compute_nhd_routing_v02(
                 for twi, (subn_tw, subn_reach_list) in enumerate(
                     reaches_ordered_bysubntw[order].items(), 1
                 ):
-                    # TODO: Confirm that a list here is best -- we are sorting,
-                    # so a set might be sufficient/better
+                    
+                    # a list of all segments in the subnetwork
                     segs = list(chain.from_iterable(subn_reach_list))
+                    
+                    # identify upstream neighbors of this subnetwork
                     offnetwork_upstreams = set()
                     segs_set = set(segs)
                     for seg in segs:
                         for us in rconn[seg]:
                             if us not in segs_set:
                                 offnetwork_upstreams.add(us)
 
+                    # add upstream neighbors to common_segs list
                     segs.extend(offnetwork_upstreams)
+                    
+                    # Define "common_segs" to identify regular routing segments
+                    common_segs = list(param_df.index.intersection(segs))
+                    
+                    # Assumes everything else is a waterbody...
+                    wbodies_segs = set(segs).symmetric_difference(common_segs)
+                    
+                    # If waterbody parameters exist
+                    if not waterbodies_df.empty:
+
+                        lake_segs = list(waterbodies_df.index.intersection(segs))
+
+                        waterbodies_df_sub = waterbodies_df.loc[
+                            lake_segs,
+                            [
+                                "LkArea",
+                                "LkMxE",
+                                "OrificeA",
+                                "OrificeC",
+                                "OrificeE",
+                                "WeirC",
+                                "WeirE",
+                                "WeirL",
+                                "ifd",
+                                "qd0",
+                                "h0",
+                            ],
+                        ]
+
+                    else:
+                        lake_segs = []
+                        waterbodies_df_sub = pd.DataFrame()
```

Comment:
addresses point 2 in https://github.com/NOAA-OWP/t-route/issues/321#issuecomment-853141327